### DISCUSSION TITLE
Chrome processes not ending

### DISCUSSION THREAD

**Discussion by phlegx** [rubycdp/ferrum#317] [General]
Hi there,

we are using Ferrum `0.13` version in our Ruby on Rails app to call HTML pages and create a screenshot of the page. However it seems that this creates  quite some never ending Chrome processes. 

The never ending chrome processes do sum up and over time eats up all the memory in our Kubernetes Containers/Pods. This is causing Kubernetes to restart the container after some time.

We can measure the amount of chrome processes that are running longer then one hour on one host with this command:

```
ps -eo etimes,pid,lstart,etime,args --sort -etimes | awk '/chrome/ {if ($1 > 1*3600) print $0 }' | wc -l
56
```

Now the question is why do these processes be there forever when they should end in less then one second usually? At least until the container is killed due to the memory limits that are set? Can this somehow be debugged?

These are the only options that we set:

```
Ferrum::Browser.new(browser_options: { 'no-sandbox':             nil,
                                   'disable-setuid-sandbox': nil,
                                   'disable-dev-shm-usage':  nil }
```

after screenshot was done we also quit the browser like:

`browser.quit`

Ideas are very welcome! Thank you in advance!

**Comment by route** [comment]
You better start one browser, and create pages in the new context, that you can dispose after job is done. Check the docs for create_page and new context option, try this approach and let me know if it’s any better

**Comment by phlegx** [comment]
We have tried to use context option with one global browser instance but the problem is still present. We also use `dumb-init` with argument `bundle exec rails s -b 0.0.0.0` in the container. What can we do?

**Reply by phlegx** [reply]
@route

**Reply by route** [reply]
@phlegx I believe that I have no info about your cluster, how you run jobs, about permissions, if your script is constantly running or one-time job and so on and so on. This is a really broad question, without knowing your arch. Try to figure out if Chrome receives any KILL signal.
